components/service/glean/src/main/java/mozilla/components/service/glean/StringMetricType.kt

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean
+
+import mozilla.components.service.glean.storages.StringsStorageEngine
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * This implements the developer facing API for recording string metrics.
+ *
+ * Instances of this class type are automatically generated by the parsers at build time,
+ * allowing developers to record values that were previously registered in the metrics.yaml file.
+ *
+ * The string API only exposes the [record] method, which takes care of validating the input
+ * data and making sure that limits are enforced.
+ */
+data class StringMetricType(
+    override val applicationProperty: Boolean = false,
+    override val disabled: Boolean = false,
+    override val category: String,
+    override val name: String,
+    override val sendInPings: List<String> = listOf("default"),
+    override val userProperty: Boolean = false,
+    val maxLength: Int
+) : CommonMetricData {
+    private val logger = Logger("glean/StringMetricType")
+
+    /**
+     * Set a string value.
+     *
+     * @param value This is a user defined string value. The maximum length of
+     * this string is defined the metrics.yaml file by max_length.
+     */
+    public fun setValue(value: String) {
```

Comment:
nit: what do you think about using simple verbs for this? for example, we could name this method `set`, similarly to the `record` from the `EventMetricType`

Reply:
Sure.  No problem.  I was going by the API docs which say `setValue`.  I can update the docs as well.